main.ipynb

```diff
@@ -666,10 +674,38 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 18,
    "id": "71f3bf15-c584-44da-a789-c821acb3f9a6",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Idx 0: Waypoint(position=(np.float32(0.0), np.float32(0.0), 30.48), speed=3.085919888690114)\n",
+      "Idx 1: Waypoint(position=(np.float32(15.167914), np.float32(0.0), 30.48), speed=3.085919888690114)\n",
```

Comment:
Please use x and y as primitive floats instead of `np.float`. It won't affect computation, but it will be easier to read.